### DISCUSSION TITLE
Using template engine with types

### DISCUSSION THREAD

**Discussion by iamguid** [orval-labs/orval#843] [Show and tell]
Hi, I just want to offer to use new template engine with types for code generation, also i can help you with maintain project. Here the article about my library https://medium.com/@iam.guid/first-typescript-jsx-based-template-engine-for-code-generation-d9be0275e6d6 what do you think?

**Comment by anymaniax** [comment]
Hello @iamguid, could you do an example with the react query generator to see the result of using a template engine?

**Reply by iamguid** [reply]
hello again, I rethought the solution and tomorrow I will write you some examples.

**Reply by iamguid** [reply]
So what do you thing about this one? https://stackblitz.com/edit/typescript-1hnpnj?file=index.ts

Templates are just functions with string template literal that get any context as parameter and return string.
Idea is simple, you register your templates in manager, then you can redefine it outside of the module and render.
After all you format it using prettier and check result code by eslint and typescript compiler (it will be part of the ready to use solution).

Pros:
1) you can redefine templates outside of module
2) you can render one template in another
3) your templates just string template literal, you don't need compiling
4) your templates are small and reusable
5) your templates are type save
6) you don't need to formatting your templates

Cons:
1) another one dependency (but it can be a part of the your library as code)

If you interesting in that solution, let me know and I will try to make a PR with more difficult example like react query generator.
